packages/sso-button-react/src/SsoButton/SsoButton.tsx

```diff
@@ -0,0 +1,147 @@
+import {useMemo} from 'react'
+import cn from 'classnames'
+import styles from './styles.module.css'
+import {Icon} from '../Icon'
+import type {LoginState, Theme, SupportedIcon} from './types'
+import {supportedIcons} from './types'
+
+const getDefaultTextContentByState = (
+  showIcons?: boolean,
+): Record<LoginState, string> => ({
+  loading: 'Connecting...',
+  error: 'Try again',
+  logged_in: 'Connected',
+  logged_out: showIcons ? 'Sign in with' : 'Sign in',
+})
+
+export type SsoButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  state: LoginState
+  showIcons?: boolean
+  textContentByState?: Partial<ReturnType<typeof getDefaultTextContentByState>>
+  theme?: Theme
+  userInfo?: {
+    name?: string
+    provider?: SupportedIcon
+  }
+}
+
+export function SsoButton({
+  state,
+  children,
+  className,
+  showIcons = true,
+  theme = 'light',
+  textContentByState = getDefaultTextContentByState(showIcons),
+  userInfo,
+  ...restProps
+}: SsoButtonProps) {
+  const textContent = useMemo(
+    () =>
+      ({
+        ...getDefaultTextContentByState(showIcons),
+        ...textContentByState,
+      }[state]),
+    [state, showIcons, textContentByState],
+  )
+
+  return (
+    <button
+      className={cn(styles.buttonReset, styles.button, className, {
+        [styles.darkMode]: theme === 'dark',
+        [styles.lightMode]: theme === 'light',
+      })}
+      {...restProps}
+      // data-error-state is used as CSS selector
+      data-error-state={state === 'error'}
```

Comment:
When adjusting button styles, there is a challenge to override specific states, like 'error' without affecting other states, and vice versa. 

Leveraging data attributes offers a clean solution. By introducing  'data-error-state='true',' we can ensure styling modifications are straightforward:

```css
/* Change color in non-error states */
.color-override {
  color: purple;
}

/* Change color only in error state */
.color-override[data-error-state='true'] {
  color: pink;
}
```